File: src/leetCode/easy/Solution27.java
package leetCode.easy;

class Solution27 {

    public static void main(String[] args) {
        Solution27 sol = new Solution27();
        int[] arr = {3, 3, 2, 2};

        sol.removeElement(arr, 3);

    }

    public int removeElement(int[] nums, int val) {
        int index = 0;
        for (int i : nums) {
            if (i != val) {
                nums[index++] = i;
            }
        }
        return index;
    }

}

File: src/leetcode/easy/Solution27.cs
// https://leetcode.com/problems/remove-element/

public class Solution27 {
    public int RemoveElement(int[] nums, int val) {

        if(nums.Length ==  0)
        {
            return 0;
        }

        int index = 0;
        int lastIndex = nums.Length - 1;
        
        while(index <= lastIndex)
        {
            if(nums[lastIndex] == val)
            {
                lastIndex--;
                continue;
            }
            if(nums[index] == val)
            {
                nums[index] = nums[lastIndex--];
            }
            index++;
        }
        return index;
    }
}
